lib/identity/stateTransitions/assetLock/proof/chain/validateChainAssetLockProofStructureFactory.js

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * @param {JsonSchemaValidator} jsonSchemaValidator
+ * @param {StateRepository} stateRepository
+ * @returns {validateChainAssetLockProofStructure}
+ */
+function validateChainAssetLockProofStructureFactory(
+  jsonSchemaValidator,
+  stateRepository,
+) {
+  /**
+   * @typedef {validateChainAssetLockProofStructure}
+   * @param {RawChainAssetLockProof} rawAssetLockProof
+   */
+  async function validateChainAssetLockProofStructure(
+    rawAssetLockProof,
+  ) {
+    const result = jsonSchemaValidator.validate(
+      chainAssetLockProofSchema,
+      convertBuffersToArrays(rawAssetLockProof),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    const {
+      coreChainLockedHeight: proofCoreChainLockedHeight,
+      outPoint: outPointBuffer,
+    } = rawAssetLockProof;
+
+    const latestPlatformBlockHeader = await stateRepository.fetchLatestPlatformBlockHeader();
+
+    const { coreChainLockedHeight: currentCoreChainLockedHeight } = latestPlatformBlockHeader;
+
+    if (currentCoreChainLockedHeight > proofCoreChainLockedHeight) {
+      result.addError(
+        new InvalidIdentityAssetLockProofHeightError(
+          proofCoreChainLockedHeight,
+          currentCoreChainLockedHeight,
+        ),
+      );
+
+      return result;
+    }
+
+    if (!await stateRepository.isAssetLockTransactionOutPointNotUsed(outPointBuffer)) {
+      result.addError(new InvalidIdentityAssetLockProofOutPointError(outPointBuffer));
+
+      return result;
+    }
+
+    const outPoint = Transaction.parseOutPointBuffer(outPointBuffer);
+    const { outputIndex } = outPoint;
+    let rawTransaction;
+
+    try {
+      rawTransaction = await stateRepository.fetchTransaction(outPoint.transactionHash);
+    } catch (e) {
+      result.addError(new InvalidIdentityAssetLockProofOutPointTransactionError(outPointBuffer));
+    }
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    /**
+     * @type {Transaction}
+     */
+    let transaction;
+    try {
+      transaction = new Transaction(rawTransaction);
+    } catch (e) {
+      const error = new InvalidIdentityAssetLockTransactionError(e.message);
+
+      result.addError(error);
+
+      return result;
+    }
+
+    if (!transaction.outputs[outputIndex]) {
+      result.addError(
+        new IdentityAssetLockTransactionOutputNotFoundError(outputIndex),
+      );
+
+      return result;
+    }
+
+    const output = transaction.outputs[outputIndex];
+
+    if (!output.script.isDataOut()) {
+      result.addError(
+        new InvalidIdentityAssetLockTransactionOutputError('Output is not a valid standard OP_RETURN output', output),
+      );
+
+      return result;
+    }
+
+    const publicKeyHash = transaction.outputs[outputIndex].script.getData();
+
+    if (publicKeyHash.length !== 20) {
+      result.addError(
+        new InvalidIdentityAssetLockTransactionOutputError('Output has invalid public key hash', output),
+      );
+
```

Comment:
Extract code to `validateAssetLockTrasaction` and use here and in `validateInstantAssetLockProofStructure`